### DISCUSSION TITLE
I uploaded my react project but link is not working.....

### DISCUSSION THREAD

**Discussion by saurav1218** [community/community#60915] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

heyy buddy, Need a help !!!!
I uploaded react files in my repo and link also generated there is a no problem in it but the main problem is after clicking in the link nothing is showing on that, I mean I uploaded every file. Followed Every steps but why It is not showing anything. 
here is the link of my repo :- https://saurav1218.github.io/hungryhub/

**Comment by jge162** [comment]
React.js isnt built for GitHub pages. Use vercel to deploy your project.

**Reply by GhostOf0days** [reply]
Or Netlify or Heroku.

**Comment by the-officialjosh** [comment]
You could use Netflify and connect your GitHub for automatic production

**Reply by GhostOf0days** [reply]
Or Heroku or Vercel.

**Comment by karakuscem** [comment]
You could use netlify, it's probably easiest solution. But if you want to use Github pages for your react project there are some steps for it.

## **Step 1: Install `gh-pages` package as a "dev-dependency" of the app**
`npm install gh-pages --save-dev`

### Step 2 : Add homepage property
Open your package.json file in your app and add `homepage` property. Define it's value to be the string `http://{username}.github.io/{repo-name}`, where `username` is your GitHub username and `{repo-name}` is the name of the Github repository it should look like this `"homepage": "http://karakuscem.github.io/react-poster"`

### Step 3 : Deploy script
In the same file (package.json) you should add two things in your existing scripts property. First one is `"predeploy": "npm run build"` and second one is `"deploy": "gh-pages -d build"`.
Your json file should look like this; 
![image](https://github.com/community/community/assets/73034223/35ceddc4-bb55-446b-acec-303708c3b49a)

Now run `npm run deploy`

### Step 4 : Setting source to gh-pages branch

After deployment, you should go to your repository on Github and click on the settings tab. Click pages and set source as gh-pages-branch.

![image](https://github.com/community/community/assets/73034223/41aa3ccc-6e59-440c-bfc6-178a5841f79f)
